File: 9041/test03/distinct_lines.pl
#!/usr/bin/perl -w



$num = $ARGV[0];
$count = 0;
%dict = ();

while (my $line = <STDIN>) {
	chomp $line;
	$line =~ s/\s//g;
	$line = lc $line;
	if ($num <= 0) {
		next;
	}
	if ($dict{$line}) {
		$count++;
		next;
	} else {
		$dict{$line} = 1;
		$num--;
		$count ++;
	}
	
}

if ($num <= 0) {
	print "$ARGV[0] distinct lines seen after $count lines read.\n";
} else {
	print "End of input reached after $count lines read - $ARGV[0] different lines not seen.\n";
}
	

File: 9041/test05/distinct_lines.py
#!/usr/bin/python3

import sys

l = sys.argv[1]
li = []
d = 0
total = 0
for line in sys.stdin:
    line = line.strip()
    new_line=""
    for w in line.split(" "):
        new_line += w.lower()
        
    total +=1
    if new_line in li:
        continue
        
    else:
        li.append(new_line)
        d += 1
        if d == int(l):
            print(l,"distinct lines seen after", total, "lines read.")
            sys.exit()
print("End of input reached after", total, "lines read - ", l, "different lines not seen.") 
        
        
    
